nixos/modules/services/torrent/deluge.nix

```diff
@@ -20,45 +44,143 @@ in {
             Number of files to allow deluged to open.
           '';
         };
+
+        config = mkOption {
+          type = types.attrs;
+          example = literalExample ''
+            {
+              download_location = "/srv/torrents/";
+              max_upload_speed = "1000.0";
+              share_ratio_limit = "2.0";
+              allow_remote = true;
+              daemon_port = 58846;
+              listen_ports = [ 6881 6889 ];
+              }
+          '';
+          description = ''
+            Deluge core configuration for the core.conf file. Only has
+            an effect when <option>services.deluge.declarative</option>
+            is set to <literal>true</literal>.
+            String values must be quoted, integer and boolean values must not.
+          '';
+        };
+
+        declarative = mkOption {
+          type = types.bool;
+          default = false;
+          description = ''
+            Whether to use a declarative deluge configuration.
+            Only if set to <literal>true</literal>, the options
+            <option>services.deluge.config</option> and
+            <option>services.deluge.auth</option> will be
+            applied.
+          '';
+        };
+
+        openFirewall = mkOption {
+          default = false;
+          type = types.bool;
+          description = ''
+            Whether to open the firewall for the ports defined in:
+            - <option>services.deluge.web.port</option>
+            - <option>services.deluge.config.daemon_port</option>
+            - <option>services.deluge.config.listen_ports</option>
+          '';
+        };
+
+        dataDir = mkOption {
+          type = types.path;
+          default = "/var/lib/deluge";
+          description = ''
+            The directory where deluge will create files.
+          '';
+        };
+
+        authFile = mkOption {
+          type = types.path;
+          example = "/run/keys/deluge-auth";
+          description = ''
+            The file managing the authentication for deluge, the format of
+            this file is straightforward, each line contains a
+            username:password:level tuple in plaintext.
+            Only has an effect when <option>services.deluge.declarative</option>
+            is set to <literal>true</literal>.
+          '';
+        };
       };
 
-      deluge.web.enable = mkEnableOption "Deluge Web daemon";
+      deluge.web = {
+        enable = mkEnableOption "Deluge Web daemon";
+
+        port = mkOption {
+        type = types.ints.u16;
+          example = 8112;
```

Comment:
You should assign 8112 as the default to this, otherwise users get an error when they just do `web.enable = true;`